Use os.CreateTemp in kubectl editor

xuhdev opened the pull request:
`os.CreateTemp` seems to perform the exactly same task here, and its
implementation seems having considered many more edge cases than the
implementation here. This patch uses `os.CreateTemp` here to avoid
reinventing the wheel.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

See the top of this comment box.

#### Does this PR introduce a user-facing change?

```release-note
None
```

dougsland reviewed (commented):
To me make sense but lets hear from others members.
/cc @eddiezane 

/priority backlog
/ok-to-test
/lgtm

dougsland commented:
/triage accepted
/priority backlog
/ok-to-test
/lgtm

eddiezane commented:
`ioutil` [is deprecated](https://golang.org/doc/go1.16#ioutil) as of Go 1.16.

Can you please swap this to [`os.CreateTemp`](https://golang.org/pkg/os/#CreateTemp)?

It looks like this code was [imported from OpenShift back in 2015](https://github.com/kubernetes/kubernetes/pull/13604) and there was a [reason](https://github.com/kubernetes/kubernetes/pull/13604/files#r38999530) for this implementation. Can you confirm that this is resolved?

/cc @smarterclayton

dougsland commented:
/lgtm cancel

dougsland reviewed (commented):
/retest

rikatz commented:
Hey @xuhdev thanks for the PR.

Overall this seems a good cleanup to me.

Will wait for some more feedback (and anyway we need to wait not until the end of code freeze).

Thanks!

/test pull-kubernetes-e2e-kind

xuhdev commented:
The test failure doesn't seem to be relevant. I rebased the PR so we can see if things change.

rikatz commented:
/test pull-kubernetes-e2e-kind

rikatz commented:
/lgtm

I guess this is a good addition at all :)

eddiezane commented on staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go:
```diff
@@ -162,27 +161,7 @@ func (e Editor) LaunchTempFile(prefix, suffix string, r io.Reader) ([]byte, stri
 }
 
 func tempFile(prefix, suffix string) (f *os.File, err error) {
```
Looks like this wrapping function isn't used elsewhere so let's remove it for extra cleanup!

xuhdev replied:
Definitely! Done

eddiezane reviewed (commented):
One more thing then lgtm!

eddiezane commented:
Please squash commits.

xuhdev committed:
Use os.CreateTemp in kubectl editor

os.CreateTemp seems to perform the exactly same task here, and its
implementation seems having considered many more edge cases than the
implementation here. This patch uses os.CreateTemp here to avoid
reinventing the wheel.

rikatz commented:
@eddiezane I've learned this trick and try to tell everyone about it:

/label tide/merge-method-squash

And now Tide will squash the commits before merging :)

eddiezane commented:
@rikatz oooo magic!

@xuhdev `-26` is a great first contribution 😄 

/approve

eddiezane commented:
@xuhdev that commit message seems to be for something else?

xuhdev commented:
Oops, I copied the wrong commit message when I was squashing 😅 Corrected now.

eddiezane commented:
Failure looks like https://github.com/kubernetes/kubernetes/issues/99801

/test pull-kubernetes-e2e-kind

rikatz commented:
/test pull-kubernetes-e2e-kind

rikatz commented:
/lgtm
Thank you @xuhdev !

dougsland reviewed (commented):
/lgtm

xuhdev commented:
Does this PR have to be milestoned?

/milestone v1.21

rikatz commented:
We're on code freeze so this will need to wait until v1.21 is released to be merged :)

xuhdev commented:
Thanks! Could someone change the milestone to v1.21? The tide bot says it won't be mergeable.

rikatz commented:
We're on code freeze so this will need to wait until v1.21 is released to be merged :) 

Once v1.21 is released, this restriction will be removed :)
